contract/models/contract.py

```diff
@@ -209,7 +209,9 @@ def _onchange_contract_template_id(self):
                     field.name in self.NO_SYNC,
                 )
             ):
-                self[field_name] = self.contract_template_id[field_name]
+                if self.contract_template_id[field_name]:
+                    self[field_name] = self.contract_template_id[field_name]
+        self._onchange_contract_type()
```

Comment:
Aren't onchange trigger in chain if executed from UI?

Reply:
`_onchange_contract_type` is called [here](https://github.com/OCA/contract/blob/12.0/product_contract/models/sale_order.py#L84)

Reply:
Yeah, but that's something that is done programmatically, not triggered by UI change.

Reply:
I change it,

`_onchange_contract_type` need to be called when creating the contract from an SO.